### PR TITLE
fix: Skip all transpilation and adding any polyfills when unknown browser version used

### PR DESCRIPTION
**Pull request by Guocork**
**Description:**

Skip all transpilation and adding any polyfills when unknown browser version used

**BREAKING CHANGE:**
I need some help. Right now, my approach is: 
when an unsupported browser version is set in browserslist-rs — for example:
```
"env": {
  "targets": ["Chrome > 130"]
}
```
Because the [`targets_to_versions`](https://github.com/swc-project/swc/blob/main/crates/swc_ecma_preset_env/src/lib.rs#L48C19-L48C38) has the  [`ignore_unkonwn_versions`](https://github.com/swc-project/swc/blob/main/crates/preset_env_base/src/query.rs#L57) option enabled in  [`browserslist_opts`](https://github.com/swc-project/swc/blob/main/crates/preset_env_base/src/query.rs#L106), it ends up returning an empty result, and [`is_any_target`](https://github.com/swc-project/swc/blob/main/crates/swc_ecma_preset_env/src/lib.rs#L49) becomes true.

So, I added a check to skip processing when the `targets` list is empty. But when I run `cargo test`, not all the tests pass. I'm not sure if there's a problem with my approach, or if I’m missing some edge cases.

**Related issue (if exists):**
#9778

### CODE REVIEW

File: crates/swc_ecma_preset_env/src/lib.rs [hunk 1]
```diff
@@ -58,7 +58,7 @@ where
             let f = transform_data::Feature::$feature;
             !exclude.contains(&f)
                 && (c.force_all_transforms
-                    || (is_any_target
+                    || (!is_any_target
```
Comment (kdy1) [inline crates/swc_ecma_preset_env/src/lib.rs:61]: `!is_any_target` is wrong way to detect unknown browser version

Comment (kdy1) [inline crates/swc_ecma_preset_env/src/lib.rs:61]: `!is_any_target` is not a way to detect an unknown browser version

**Review by kdy1 — CHANGES_REQUESTED** [review]
image

You have to look at failing test cases. You would be able to adjust the condition to create the correct condition for the input case.
